sfyra/pkg/tests/server.go

```diff
@@ -259,6 +259,127 @@ func TestServerAcceptance(ctx context.Context, metalClient client.Client, vmSet
 	}
 }
 
+// TestServerPaused makes sure the paused bool works.
+func TestServerPaused(ctx context.Context, metalClient client.Client, vmSet *vm.Set) TestFunc {
+	return func(t *testing.T) {
+		const numDummies = 3
+
+		// create dummy servers to test with
+		dummySpec := v1alpha1.ServerSpec{
+			CPU: &v1alpha1.CPUInformation{
+				Manufacturer: "DummyManufacturer",
+			},
+		}
+
+		for i := 0; i < numDummies; i++ {
+			serverName := fmt.Sprintf("dummyserver-%s", strconv.Itoa(i))
+			_, err := createDummyServer(ctx, metalClient, serverName, dummySpec)
+			require.NoError(t, err)
+		}
+
+		dummyServers := &v1alpha1.ServerList{}
+
+		labelSelector, err := labels.Parse("dummy-server=")
+		require.NoError(t, err)
+
+		// wait for all the servers to be registered
+		require.NoError(t, retry.Constant(1*time.Minute, retry.WithUnits(10*time.Second)).Retry(func() error {
+			err = metalClient.List(ctx, dummyServers, client.MatchingLabelsSelector{Selector: labelSelector})
+			if err != nil {
+				return err
+			}
+
+			if len(dummyServers.Items) != numDummies {
+				return retry.ExpectedError(fmt.Errorf("%d != %d", len(dummyServers.Items), numDummies))
+			}
+
+			return nil
+		}))
+
+		// clean up dummies
+		defer func(client client.Client) {
+			for _, server := range dummyServers.Items {
+				server := server
+				client.Delete(ctx, &server)
+			}
+		}(metalClient)
+
+		// patch all servers as accepted
+		for _, server := range dummyServers.Items {
+			server := server
+
+			patchHelper, err := patch.NewHelper(&server, metalClient)
+			require.NoError(t, err)
+
+			server.Spec.Accepted = true
+			require.NoError(t, patchHelper.Patch(ctx, &server))
+		}
+
+		// verify that all servers shows up as available in `any` serverclass
+		require.NoError(t, retry.Constant(1*time.Minute, retry.WithUnits(10*time.Second)).Retry(func() error {
+			var serverClass v1alpha1.ServerClass
+			err := metalClient.Get(ctx, types.NamespacedName{Name: v1alpha1.ServerClassAny}, &serverClass)
+			if err != nil {
+				return err
+			}
+			if len(serverClass.Status.ServersAvailable) != numDummies {
```

Comment:
this is a wrong check: there might be other `Server`s left from previous runs, so we should better check that `ServerClass` explicitly lists dummies' IDs in the list as 'Available'

Reply:
fixed